### PR TITLE
Fix DLC not storing nSequence for funding inputs

### PR DESCRIPTION
**Pull request by benthecarman**
Started out trying to make it default to signaling for RBF for DLCs, ended up finding out that we assumed a sequence for every input. Fixed that and added the new default so we get better privacy benefits.

This way we will by default share the anon set with dual funded lightning txs, as they appear to be using an nSequence of `0xfffffffd`

https://mempool.space/tx/91538cbc4aca767cb77aa0690c2a6e710e095c8eb6d8f73d53a3a29682cb7581

### CODE REVIEW

**Review by nkohen — APPROVED** [review]

**Comment by Christewart** [comment]
CI is being weird here, not sure why it is failing. 

Is there any implications here with the spec? I.e. if CG or AF didn't upgrade, we can still use them right?

**Comment by benthecarman** [comment]
> CI is being weird here, not sure why it is failing.

Just updated description, was a bug in our code :(

> Is there any implications here with the spec? I.e. if CG or AF didn't upgrade, we can still use them right?

Spec allows for any sequence number, other clients should be compliant unless they have the same bug we did

File: dlc-test/src/test/scala/org/bitcoins/dlc/testgen/TestVectorGen.scala [hunk 1]
```diff
@@ -36,6 +36,12 @@ trait TestVectorGen[T <: TestVector, Input] {
   }
 
   def readFromDefaultTestFile(): JsResult[Vector[T]] = {
+    // If we are missing the test file, generate it
+    if (!defaultTestFile.canRead) {
+      val gen = generateAndWriteTestVectors()
+      Await.result(gen, 60.seconds)
```
Comment (Christewart) [inline dlc-test/src/test/scala/org/bitcoins/dlc/testgen/TestVectorGen.scala:42]: Any reason to not propagate the Future?

**Review by Christewart — APPROVED** [review]

File: dlc-test/src/test/scala/org/bitcoins/dlc/testgen/TestVectorGen.scala (hunk 1 above)
Comment (benthecarman) [reply dlc-test/src/test/scala/org/bitcoins/dlc/testgen/TestVectorGen.scala:42]: Would have been required a much larger refactor, probably better for a separate PR

**Pull request merged**